ssl/ssl_asn1.cc

```diff
@@ -841,20 +842,14 @@ int i2d_SSL_SESSION(SSL_SESSION *in, uint8_t **pp) {
   if (!SSL_SESSION_to_bytes(in, &out, &len)) {
     return -1;
   }
+  bssl::UniquePtr<uint8_t> free_out(out);
 
-  if (len > INT_MAX) {
-    OPENSSL_free(out);
-    OPENSSL_PUT_ERROR(SSL, ERR_R_OVERFLOW);
-    return -1;
-  }
-
-  if (pp) {
-    OPENSSL_memcpy(*pp, out, len);
-    *pp += len;
+  ScopedCBB cbb;
+  if (!CBB_init(cbb.get(), 256) || !CBB_add_bytes(cbb.get(), out, len)) {
```

Comment:
do we need to advance `*pp`?

```
 // If |outp| and |*outp| are non-NULL, the function writes the result to 
 // |*outp|, which must have enough space available, and advances |*outp| just 
 // past the output. 
```

Reply:
Yes the very useful `CBB_finish_i2d` happens to handle all of the `i2d` shenanigans for us :) 

https://github.com/aws/aws-lc/blob/65d8c946b7ec57d258f35158d02d0f6f90c86eff/crypto/bytestring/internal.h#L60-L67

Reply:
nice. looking back i see we assert that on L2040 of the test.